Fix some Chinese translation errors and add utf-8 support

Caffreyfans opened the pull request.

Caffreyfans committed:
add Chinese translate

crankyoldgit committed:
Fix linter and syntax errors

* Linter issues:
```
src/locale/zh-CN.h:4:  #ifndef header guard has wrong style, please use: LOCALE_ZH_CN_H_  [build/header_guard] [5]
src/locale/zh-CN.h:463:  #endif line should be "#endif  // LOCALE_ZH_CN_H_"  [build/header_guard] [5]
src/locale/zh-CN.h:110:  Multi-line string ("...") found.  This lint script doesn't do well with such strings, and may give bogus warnings.  Use C++11 raw strings or concatenation instead.  [readability/multiline_string] [5]
src/locale/zh-CN.h:296:  Multi-line string ("...") found.  This lint script doesn't do well with such strings, and may give bogus warnings.  Use C++11 raw strings or concatenation instead.  [readability/multiline_string] [5]
src/locale/zh-CN.h:423:  Multi-line string ("...") found.  This lint script doesn't do well with such strings, and may give bogus warnings.  Use C++11 raw strings or concatenation instead.  [readability/multiline_string] [5]
```
* Fix some extraneous '"' characters.
* Disable `D_STR_THREELETTERDAYS` as it really needs to be three character per day.

crankyoldgit committed:
Fix copyright date

crankyoldgit committed:
Add chinese localisation to IRrecvDumpV2 build options

Caffreyfans committed:
Fix translation errors

Caffreyfans committed:
add utf-8 support

Caffreyfans committed:
add utf-8 support

Caffreyfans committed:
Merge branch 'master' into master

crankyoldgit commented on examples/IRMQTTServer/IRMQTTServer.ino:
```diff
@@ -1001,7 +1001,7 @@ String htmlSelectSwingh(const String name, const stdAc::swingh_t def) {
   String html = F("<html><head><title>");
   html += title;
-  html += F("</title></head><body><center><h1>");
+  html += F("</title><meta http-equiv=\"Content-Type\"content=\"text/html;charset=utf-8\"></head><body><center><h1>");
```
Linter error:
```
examples/IRMQTTServer/IRMQTTServer.ino:1004:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
```

crankyoldgit replied:
Linter error:
```
examples/IRMQTTServer/IRMQTTServer.ino:1004:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
```

See: https://travis-ci.org/github/crankyoldgit/IRremoteESP8266/jobs/676284478#L632

Also, please add a space between `http-equiv=` & `content=`
e.g.
...`http-equiv=\"Content-Type\" content=\"`...

crankyoldgit commented on examples/IRServer/IRServer.ino:
```diff
@@ -65,7 +65,9 @@ IRsend irsend(kIrLed);  // Set the GPIO to be used to sending the message.
-                "<head><title>" HOSTNAME " Demo</title></head>" \
+                "<head><title>" HOSTNAME " Demo </title>"\
+                "<meta http-equiv=\"Content-Type\"content=\"text/html;charset=utf-8\"> \
+                </head>" \
```
Linter errors:
```
examples/IRServer/IRServer.ino:69:  Multi-line string ("...") found.  This lint script doesn't do well with such strings, and may give bogus warnings.  Use C++11 raw strings or concatenation instead.  [readability/multiline_string] [5]
examples/IRServer/IRServer.ino:69:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
examples/IRServer/IRServer.ino:69:  Missing spaces around =  [whitespace/operators] [4]
examples/IRServer/IRServer.ino:69:  Missing spaces around <  [whitespace/operators] [3]
examples/IRServer/IRServer.ino:69:  Missing space after ;  [whitespace/semicolon] [3]
examples/IRServer/IRServer.ino:70:  Multi-line string ("...") found.  This lint script doesn't do well with such strings, and may give bogus warnings.  Use C++11 raw strings or concatenation instead.  [readability/multiline_string] [5]
```

See: https://travis-ci.org/github/crankyoldgit/IRremoteESP8266/jobs/676284478#L639-L644

crankyoldgit replied:
Ditto the space between http-equiv and content per earlier comment.

crankyoldgit commented on src/locale/zh-CN.h:
```diff
@@ -329,7 +329,8 @@
 #ifndef D_STR_WIDE
-#define D_STR_WIDE "Wide"
+#define D_STR_WIDE "扫风"
+
```
Remove this added blank line.

crankyoldgit reviewed (changes requested):
Thanks for this. It looks mostly good. Just some automatically found linter/style issues to fix.

crankyoldgit committed:
Address linter issues in IRserver.ino

```
examples/IRServer/IRServer.ino:69:  Multi-line string ("...") found.  This lint script doesn't do well with such strings, and may give bogus warnings.  Use C++11 raw strings or concatenation instead.  [readability/multiline_string] [5]
examples/IRServer/IRServer.ino:69:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
examples/IRServer/IRServer.ino:69:  Missing spaces around =  [whitespace/operators] [4]
examples/IRServer/IRServer.ino:69:  Missing spaces around <  [whitespace/operators] [3]
examples/IRServer/IRServer.ino:69:  Missing space after ;  [whitespace/semicolon] [3]
examples/IRServer/IRServer.ino:70:  Multi-line string ("...") found.  This lint script doesn't do well with such strings, and may give bogus warnings.  Use C++11 raw strings or concatenation instead.  [readability/multiline_string] [5]
```

crankyoldgit committed:
Remove extraneous blank line

crankyoldgit committed:
Fix typo

crankyoldgit committed:
fix lint error

crankyoldgit committed:
Fix linter error

Hopefully.

crankyoldgit reviewed (approved):
I submitted changes to fix the linter issues etc.

Caffreyfans commented:
> I submitted changes to fix the linter issues etc.

Thanks for merging

crankyoldgit commented:
FYI, this has been included in the latest release of the library.
